### PR TITLE
[REVIEW] Remove pg dependency from cugraph store.py

### PR DESCRIPTION
**Pull request by VibhuJawa**
To enable using the same functions on both remote graph and local graph (See PR: https://github.com/rapidsai/cugraph/pull/2850) I have reformatted the class to remove dependency from  PropertyGraph, cugraph, cudf, dask_cudf 

CC: @rlratzel , @alexbarghi-nv

### CODE REVIEW

File: python/cugraph/cugraph/gnn/dgl_extensions/cugraph_store.py [hunk 1]
```diff
@@ -46,10 +34,13 @@ class CuGraphStore:
...
 
     def __init__(self, graph, backend_lib="torch"):
-        if isinstance(graph, (PropertyGraph, MGPropertyGraph)):
+
+        if type(graph).__name__ in ["PropertyGraph", "MGPropertyGraph"]:
             self.__G = graph
         else:
             raise ValueError("graph must be a PropertyGraph or MGPropertyGraph")
+
+        BaseCuGraphStore.__init__(self, graph)
```
Comment (rlratzel) [inline python/cugraph/cugraph/gnn/dgl_extensions/cugraph_store.py:43]: I think this is usually preferred:
```suggestion
        super().__init__(graph)
```

File: python/cugraph/cugraph/gnn/dgl_extensions/feature_storage.py [hunk 2]
```diff
@@ -36,7 +35,7 @@ def __init__(
             from cupy import from_dlpack
         else:
             raise NotImplementedError(
-                f"Only PyTorch ('torch'), TensorFlow ('tf'), and CuPy ('cupy') "
+                f"Only PyTorch ('torch'), TensorFlow ('tf'), and CuPy ('cupy')"
```
Comment (rlratzel) [inline python/cugraph/cugraph/gnn/dgl_extensions/feature_storage.py:38]: isn't the trailing space needed?

File: python/cugraph/cugraph/gnn/dgl_extensions/feature_storage.py [hunk 3]
```diff
@@ -67,7 +66,7 @@ def fetch(self, indices, device=None, pin_memory=False, **kwargs):
         # Default implementation uses synchronous fetch.
 
         indices = cp.asarray(indices)
-        if isinstance(self.pg, MGPropertyGraph):
+        if type(self.pg).__name__ in "MGPropertyGraph":
```
Comment (rlratzel) [inline python/cugraph/cugraph/gnn/dgl_extensions/feature_storage.py:69]: I think this is a potential subtle bug, or did you mean to use `in`?  Here's an example:
```python
>>> x=2
>>> type(x).__name__ in "int"
True
>>> type(x).__name__ in "iiinttt"
True
>>> type(x).__name__ == "int"
True
>>> type(x).__name__ == "iiinttt"
False
```

**Review by rlratzel — COMMENTED** [review]
looks good, just a few suggstions/questions.

File: python/cugraph/cugraph/gnn/dgl_extensions/feature_storage.py (hunk 3 above)
Comment (rlratzel) [reply python/cugraph/cugraph/gnn/dgl_extensions/feature_storage.py:69]: A more relevant example :)
```python
>>> pg=PropertyGraph()
>>> type(pg).__name__ in "MGPropertyGraph"
True
```

Comment (VibhuJawa) [reply python/cugraph/cugraph/gnn/dgl_extensions/feature_storage.py]: Resolved

File: python/cugraph/cugraph/gnn/dgl_extensions/feature_storage.py (hunk 2 above)
Comment (VibhuJawa) [reply python/cugraph/cugraph/gnn/dgl_extensions/feature_storage.py]: Resolved

**Review by rlratzel — APPROVED** [review]

**Comment by rlratzel** [comment]
@gpucibot merge

**Comment by VibhuJawa** [comment]
@gpucibot  rerun

**Comment by VibhuJawa** [comment]
@gpucibot rerun tests

**Comment by BradReesWork** [comment]
rerun tests

**Comment by BradReesWork** [comment]
rerun tests

**Comment by BradReesWork** [comment]
rerun tests

**Comment by VibhuJawa** [comment]
rerun tests

**Comment by VibhuJawa** [comment]
rerun tests

**Pull request merged**